src/components/Waypoints/WaypointMenuEdit.tsx

```diff
@@ -0,0 +1,174 @@
+import React, { useState, FormEventHandler } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { Input } from "@material-tailwind/react";
+
+import { XMarkIcon, PaperAirplaneIcon } from "@heroicons/react/24/outline";
+
+import type { Waypoint } from "@bindings/protobufs/Waypoint";
+
+import { deviceSliceActions } from "@features/device/deviceSlice";
+import { requestNewWaypoint } from "@features/device/deviceActions";
+import { selectActiveWaypoint } from "@app/features/device/deviceSelectors";
+
+import { useToggleEditWaypoint } from "@app/utils/waypoint";
+
+// This component contains the Waypoint Menu when it is not being edited
+// It is called in MapView.tsx
+
+const WaypointMenuEdit = () => {
+  const dispatch = useDispatch();
+  const activeWaypoint = useSelector(selectActiveWaypoint());
+
+  // Waypoint info
+  const [waypointTitle, setWaypointTitle] = useState(activeWaypoint?.name);
+  const [waypointDescription, setWaypointDescription] = useState(
+    activeWaypoint?.description
+  );
+  const waypointLat = activeWaypoint?.latitudeI
+    ? (activeWaypoint.latitudeI / 1e7).toString()
+    : null;
+  const waypointLong = activeWaypoint?.longitudeI
+    ? (activeWaypoint.longitudeI / 1e7).toString()
+    : null;
+
+  // Utils
+  const toggleEditWaypoint = useToggleEditWaypoint();
+  const handleSubmit: FormEventHandler = (e) => {
+    e.preventDefault();
+    console.log("Clicked send");
+
+    const channelNum =
+      Number((document.getElementById("channel") as HTMLInputElement).value) ??
+      0;
+
+    if (activeWaypoint) {
+      const updatedWaypoint: Waypoint = {
+        ...activeWaypoint,
+        name: waypointTitle ? waypointTitle : "",
+        description: waypointDescription ? waypointDescription : "",
+      };
+      dispatch(
+        requestNewWaypoint({ waypoint: updatedWaypoint, channel: channelNum })
+      );
+      dispatch(deviceSliceActions.setWaypointEdit(false));
+    } else {
+      console.log("Error: No active waypoint");
+    }
+  };
+
+  // Only display if there is a selected waypoint
+  if (!activeWaypoint) {
+    return <></>;
```

Comment:
See comment above, I would also recommend returning `null` instead of a fragment